core/kazoo_number_manager/src/knm_port_request.erl

```diff
@@ -506,10 +532,15 @@ migrate() ->
 %% @doc
 %% @end
 %%------------------------------------------------------------------------------
--spec completed_port(kz_json:object()) -> transition_response().
-completed_port(PortReq) ->
-    lager:debug("transitioning numbers to active"),
-    transition_numbers(PortReq).
+-spec completed_port(kz_json:object(), kz_json:object()) -> transition_response().
+completed_port(PortReq, JObj) ->
+    AccountId = kz_doc:account_id(PortReq),
+    Numbers = kz_json:get_keys(kzd_port_requests:numbers(PortReq)),
+    lager:debug("transitioning numbers ~p to active", [Numbers]),
+    {'ok', App} = maybe_reconcile_app_used_by(lists:last(Numbers), AccountId, JObj),
+    Rc = transition_numbers(PortReq),
```

Comment:
Since we can set the `used_by` at the number creation, then just call `transition_numbers(PortReq)` and if it `{ok, Save}` then process and assigned_to_app them